### ISSUE TITLE
Telnet Connection Problems

### ISSUE THREAD

**Issue by sachasmart** [stiliajohny/Bebop-On-Off-Button#1]
Apologizes that this may be out of the blue. I am very interested in getting the GPS data from my Bebop 1 and mapping in a GIS tool. I understand that this [could be possible](https://parrotpilots.com/threads/gps-and-return-home-with-bebop-1.3295/). 

I am trying to connect to my Bebop over telnet. However, when I turn on my Drone with 4 clicks on the power button, telnet is not enabled. Furthermore, when I run a port scan (`nmap 192.168.42.1 -Pn`), I only have port 21 (ftp) open. I can confirm that I can connect over ftp. 

I really appreciate your scripts and this [guide](https://fargesportfolio.com/wp-content/uploads/2018/01/BeebopHackingGuide1_7_2.pdf) has been extremely helpful. I suppose a good starting point would be to see which ARDrone version I am running, but I cannot see a `/data/system.conf` when I connect over ftp - just the `internal_000` directory with this structure:

```
.
├── Bebop_Drone
│   ├── academy
│   ├── media
│   ├── navdata
│   └── thumb
├── Debug
│   ├── archive
│   ├── crash_reports
│   └── current
├── flightplans
├── log
├── lost+found
│   └── try_create
└── scripts
```

Any suggestions - again really appreciate all the help.

**Comment by sachasmart** [comment]
Update: 

I managed to connect to my Bebop over telnet - my mistake was I needed to power on the Bebop and then press the button four times. 

I have followed your `README.md` and added your scripts to the `/bin/onoffbutton`. However, after `chmod`ing and `sync;reboot` the telnet commands, nor the factory reset long button hold works. 

My Bebop still flies and such, however I no longer have telnet access as I the four button presses do not work. 

Please - any suggestions? I tried dropping the firmware file `bebop2_update_v_4_0_4.plf` with `ftp 192.168.42.1 51` which did not work (confused about this as well). 

Upon inspection of my terminal logs it looks like the very_long button press from this repo may have overriden (likely did) the the default very long press shell script. It also looks like the scripts are not executable either, even though I did chmod 755 them. Not sure if there's a work around this other than changing the firmware or connecting via an FTDI cable - thoughts?

I did luckily make a copy of the `/bin/onoffbutton` before copying over your files (`/bin/onoffbutton_backup`) is there a way I can move these back to their original location without telnet access?


Update: I've connected with an FTDI cable and when I hold the long press to do a factory reset I get:

```
boxinit: untracked pid 1245 'longpress_0.sh' exited <--- interesting as this should be verylongpress_0.sh 
boxinit: untracked pid 1244 'BLDC_Test_Bench' exited
```
And still nothing happens :(. And none of the other short/long presses register in the tty session. I think I'm potentially screwed on the telnet side (would really appreciate some help). Looks like I can't write commands with the existing FTDI connect (`screen /dev/cu.usbserial-AL03PD04 115200`) and it could possibly have been disabled.

**Comment by sachasmart** [comment]
I managed to solve this by enabling UART commands with the [firmware linked]([here](https://github.com/Parrot-Developers/ardupilot/releases/download/bebop-v0.0/bebopdrone_update.plf)) in this [documentation]([here](https://github.com/Parrot-Developers/ardupilot/releases/download/bebop-v0.0/bebopdrone_update.plf). 

This changed my firmware to `0.0.0`, but allowed me to run commands through UART. I changed around the boot up scripts, which allowed me to then factory reset my Bebop. I then had to also upgrade my drone to `4.0.5` so I could get my video connection (RDP I believe) the the FreeFlight app. 

Then my motors weren't working as expected and the drone kept flipping (sigh, never ending! I just want GPS data lol). So after tinkering with that I default to yet another upgrade to Parrot's `4.0.6` firmware, and dropped that onto the `internal_000` and rebooted. This seems to have everything back to normal. 

Lessons learned:
Always make privileges are correct and always open up port 71 for ftp so files can be moved around if telnet on 23 doesn't work. 
```
71 stream tcp nowait root ftpd ftpd -SS /
```

What a day - closing this and I hope this helps someone :)
